backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -264,5 +264,17 @@ bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
     return false;
 }
 
+// Store the user metadata structure in 'structure->mdStruct' which will be
+// later used for offset value calculation for metadata field
+bool CollectUserMetadataStructure::preorder(const IR::DpdkAsmProgram *p) {
+    for (auto st : p->structType) {
+        if (isMetadataStruct(st)) {
+            structure->mdStruct = st;
+            break;
+        }
+    }
+    return true;
```

Comment:
This should be returning false. You don't have any other visitors.

Reply:
updated